auacm/app/modules/competition_manager/views.py

```diff
@@ -181,3 +202,129 @@ def get_competition_data(cid):
         'compProblems': comp_problems,
         'teams': scoreboard
     })
+
+
+@app.route('/api/competitions/<int:cid>/register', methods=['POST'])
+@login_required
+def register_for_competition(cid):
+    """ Called when a user wants to register for a competition.
+
+    All the user has to do is submit a post to this url with no form data.
+    From their logged-in status, we'll go ahead and add them to the competiton
+    as an individual (team name is default their display name).
+    """
```

Comment:
Can this say something about how it works with an admin? I'd like to have it in writing.